core/trino-spi/src/main/java/io/trino/spi/connector/InMemoryRecordSet.java

```diff
@@ -68,15 +68,15 @@ public RecordCursor cursor()
         return new InMemoryRecordCursor(types, records.iterator());
     }
 
-    private static class InMemoryRecordCursor
+    public static class InMemoryRecordCursor<T extends Iterator<? extends List<?>>>
```

Comment:
The class became generic, but so there are now warnings wherever it's used as a raw class (eg 3 lines above).

BTW it seems the generics are not leveraged, since T shows up only in the constructor declaration.
https://github.com/trinodb/trino/pull/16295